File: salary2/AddSalariedEmployee.java
class AddSalariedEployee extends AddEmployeeTransaction {
  private double itsSalary;

  public AddSalariedEployee(int empId, String name, String address, double salary) {
    super(empId, name, address);
    itsSalary = salary;
  }

  public PaymentClassification GetClassification() {
    return new SalariedClassification(itsSalary);
  }

  public PaymentSchedule GetSchedule() {
    return new MonthlySchedule();
  }
}

File: salary/lib/add_salaried_employee.rb
require 'add_employee_transaction'

class AddSalariedEmployee < AddEmployeeTransaction
  
  attr_accessor :salary
  
  def initialize(employee_id, name, address, salary)
    super(employee_id, name, address)
    @salary = salary
  end
  
  def classification
    SalariedClassification.new(@salary)
  end
  
  def schedule
    MonthlySchedule.new
  end
end
